crypto/s2n_cipher.h

```diff
@@ -82,7 +82,7 @@ struct s2n_cipher {
         struct s2n_composite_cipher comp;
     } io;
     uint8_t key_material_size;
-    uint8_t (*is_available)(void);
+    S2N_RESULT (*is_available)(bool *available);
     int (*init)(struct s2n_session_key *key);
```

Comment:
This is a tricky one. While our general guidance is that all methods should return S2N_RESULT, it can make boolean checks like this awkward. I would think about whether we actually care about the distinction between is_available failing and is_available returning false. 

I would guess that we don't care, so my suggestion would be to use the signature
```suggestion
    bool (*is_available)(void);
    int (*init)(struct s2n_session_key *key);
```
That should probably simplify this diff too. is_available was already being treated like it returned bool, it just didn't.

If any of the implementations of is_available are complex / require error checking, you can structure them like:
```
static S2N_RESULT s2n_assert_is_available() {
    RESULT_GUARD_POSIX(blah);
    RESULT_ENSURE(blah);
    return S2N_RESULT_OK;
}

bool s2n_is_available() {
    return s2n_result_is_ok(s2n_assert_is_available());
}
```